### ISSUE TITLE
Upgrading to Apollo Server 3.x

### ISSUE THREAD

**Issue by abdavid** [moleculerjs/moleculer-apollo-server#100]
Are there plans in motion to upgrade to latest Apollo server? 

One of the main breaking changes revolve around subscriptions being refactored out. The [migration guide](https://www.apollographql.com/docs/apollo-server/migration/) outlines a possible way to continue to support subscriptions through `ws` with a workaround, but also states that client would have to be adjusted accordingly to make it work as I understand it. 

I did a quick refactor in my fork of this project to see how much would break, and while the subscription support has been removed and some of the exports from `apollo-core-server` has changed, I managed to get tests to pass (while still removing references to subscriptions though!)

Thinking ahead would it be feasible to create a new major version of this module, documenting the breaking changes so that those that require the subscription support can stay in 2.x land, while those who do not get the bright and shiny 3.x, and start paving the way for 3.x that will get subscription support at some point? 

Thoughts? @icebob

**Comment by icebob** [comment]
Yeah, I saw the migration guide and I saw it contains a lot of breaking changes. By the way, would be good to upgrade to the latest Apollo Server but would be good for any help. Could you open a PR with your changes?

**Comment by abdavid** [comment]
I would love to help out. Ill create a PR with my changes later today, would probably need to clean it up since I mostly just focused on finding the pieces that broke and getting the tests to pass.

**Comment by HazemKhaled** [comment]
@abdavid how is the progress from your side? Create a draft if you need hands

**Comment by Rangyia** [comment]
@icebob, Is there an update regarding this, or a projected timeline?

**Comment by veeramarni** [comment]
@icebob there is an existing PR, can it be merged?

**Comment by icebob** [comment]
Not yet, because the PR is not correct.

**Comment by jbool24** [comment]
> [v3 officially deprecated](https://www.apollographql.com/docs/apollo-server/previous-versions)

So, now should we skip v3 and try upgrading all deps to be compatible with v4?? How can I help??

**Comment by AndreMaz** [comment]
@jbool24 please check this PR. It already contains some work re migration to v4. Please check this for more context https://github.com/moleculerjs/moleculer-apollo-server/pull/118#issuecomment-1597571741

**Comment by shawnmcknight** [comment]
I've also created a package which eliminates apollo server entirely in favor of graphql-tools. It needs some polish still but it's working. I don't have a timeline for doing the additional work, but I think this path is better than relying on Apollo server. https://github.com/shawnmcknight/moleculer-graphql

**Comment by abdavid** [comment]
Thanks @shawnmcknight ! Ill test out replacing our current apollo variant with `@shawnmcknight/moleculer-graphql` 👍 as for this repo here, if someone really needs apollo I think they would be better off re-implementing this from scratch.

**Comment by abdavid** [comment]
@shawnmcknight more or less a complete drop-in replacement, thanks for your awesome work. Though I get what you mean  that it needs a little polish, some types more optional types and more examples will go along way.
